### PR TITLE
 fix bug for query when value is null

### PR DESCRIPTION
**Pull request by qwdarrenyin**
https://github.com/FISCO-BCOS/AMDB/pull/58

fix bug for query when value is null

### CODE REVIEW

File: libstorage/SQLBasicAccess.cpp [hunk 1]
```diff
@@ -75,18 +74,18 @@ int SQLBasicAccess::Select(h256, int64_t, const std::string& _table, const std::
...
             {
-                value.push_back(ResultSet_getString(result, index));
+                if (ResultSet_getString(result, index) != NULL)
+                {
+                    value[ResultSet_getColumnName(result, index)] =
+                        ResultSet_getString(result, index);
+                }
             }
-            valueList.push_back(value);
+            values.push_back(value);
```
Comment (wheatli) [inline libstorage/SQLBasicAccess.cpp]: it is costly in this way, you should create a map and push it into `values` before assign

File: libstorage/ZdbStorage.cpp [hunk 2]
```diff
@@ -52,30 +50,32 @@ Entries::Ptr ZdbStorage::select(h256 _hash, int64_t _num, TableInfo::Ptr _tableI
...
+    auto it = values.begin();
+    for (; it != values.end(); ++it)
     {
         Entry::Ptr entry = std::make_shared<Entry>();
-        for (size_t j = 0; j < it->size(); ++j)
+        auto it2 = it->begin();
+        for (; it2 != it->end(); ++it2)
         {
-            if (columns[j] == ID_FIELD)
+            if (it2->first == ID_FIELD)
```
Comment (bxq2011hust) [inline libstorage/ZdbStorage.cpp]: https://github.com/FISCO-BCOS/FISCO-BCOS/blob/623ba91f99ae7fab08971618378a74094dbd634d/libstorage/RocksDBStorage.cpp#L79-L85

can we do same as RocksDBStorage ?

File: libstorage/SQLStorage.cpp [hunk 3]
```diff
@@ -107,48 +114,87 @@ Entries::Ptr SQLStorage::select(h256 hash, int64_t num, TableInfo::Ptr tableInfo
...
+                Entry::Ptr entry = std::make_shared<Entry>();
+
+                Json::Value::Members member = line.getMemberNames();
+                for (Json::Value::Members::iterator iter = member.begin(); iter != member.end();
+                     ++iter)
                 {
-                    entry->setStatus(fieldValue);
+                    std::string key = *iter;
+                    std::string fieldValue = line.get(key, "").asString();
+                    if (key == ID_FIELD)
```
Comment (bxq2011hust) [inline libstorage/SQLStorage.cpp]: https://github.com/FISCO-BCOS/FISCO-BCOS/blob/623ba91f99ae7fab08971618378a74094dbd634d/libstorage/RocksDBStorage.cpp#L79-L85

can we do same as RocksDBStorage ?

File: libstorage/SQLStorage.cpp [hunk 4]
```diff
@@ -107,48 +114,87 @@ Entries::Ptr SQLStorage::select(h256 hash, int64_t num, TableInfo::Ptr tableInfo
...
+                Json::Value line = responseJson["result"]["data"].get(i, "");
+                Entry::Ptr entry = std::make_shared<Entry>();
 
-                if (columns[j] == ID_FIELD)
+                for (Json::ArrayIndex j = 0; j < line.size(); ++j)
                 {
-                    entry->setID(fieldValue);
+                    std::string fieldValue = line.get(j, "").asString();
+
+                    if (columns[j] == ID_FIELD)
```
Comment (bxq2011hust) [inline libstorage/SQLStorage.cpp]: https://github.com/FISCO-BCOS/FISCO-BCOS/blob/623ba91f99ae7fab08971618378a74094dbd634d/libstorage/RocksDBStorage.cpp#L79-L85

can we do same as RocksDBStorage ?

**Pull request merged**